

On ia64:

arch/ia64/kernel/irq.c:625: warning: implicit declaration of function `setup_irq'
arch/ia64/kernel/irq_ia64.c:237: warning: implicit declaration of function `setup_irq'
arch/ia64/kernel/mca.c:1397: warning: implicit declaration of function `setup_irq'


Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/include/linux/irq.h |    3 ++-
 1 files changed, 2 insertions(+), 1 deletion(-)

diff -puN include/linux/irq.h~setup_irq-warning-fixes include/linux/irq.h
--- 25/include/linux/irq.h~setup_irq-warning-fixes	Mon Oct  4 13:23:31 2004
+++ 25-akpm/include/linux/irq.h	Mon Oct  4 13:23:54 2004
@@ -72,6 +72,8 @@ extern irq_desc_t irq_desc [NR_IRQS];
 
 #include <asm/hw_irq.h> /* the arch dependent stuff */
 
+extern int setup_irq(unsigned int irq, struct irqaction * new);
+
 #ifdef CONFIG_GENERIC_HARDIRQS
 extern cpumask_t irq_affinity[NR_IRQS];
 
@@ -81,7 +83,6 @@ extern asmlinkage unsigned int __do_IRQ(
 extern void note_interrupt(unsigned int irq, irq_desc_t *desc, int action_ret);
 extern void report_bad_irq(unsigned int irq, irq_desc_t *desc, int action_ret);
 extern int can_request_irq(unsigned int irq, unsigned long irqflags);
-extern int setup_irq(unsigned int irq, struct irqaction * new);
 
 extern void init_irq_proc(void);
 #endif
_
